### PR TITLE
CHE-367: Return websocket URL of extension server when a workspace is launched

### PR DESCRIPTION
**Pull request by mmorhun**

### CODE REVIEW

**Comment by mmorhun** [comment]
@garagatyi, check please

**Comment by garagatyi** [comment]
have you checked that build/tests/IDE is working fine after your changes?

File: core/platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java [hunk 1]
```diff
@@ -954,7 +958,27 @@ private void ensureUserIsWorkspaceOwner(UsersWorkspace usersWorkspace) throws Se
...
-                                 Constants.STOP_WORKSPACE));
+                                 STOP_WORKSPACE));
+
+            if (RuntimeWorkspaceDto.class.isAssignableFrom(workspace.getClass())) {
+                RuntimeWorkspaceDto runtimeWorkspace = (RuntimeWorkspaceDto)workspace;
+                String uri = null;
+                for (ServerDto server : runtimeWorkspace.getDevMachine()
+                                                        .getRuntime()
+                                                        .getServers()
+                                                        .values()) {
```
Comment (garagatyi) [inline core/platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java]: use java 8 stream syntax instead

File: core/platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java [hunk 2]
```diff
@@ -954,7 +958,27 @@ private void ensureUserIsWorkspaceOwner(UsersWorkspace usersWorkspace) throws Se
...
+                    if (WS_AGENT_SERVER_REFERENCE.equals(server.getRef())) {
+                        uri = server.getUrl();
+                        break;
+                    }
+                }
+                links.add(createLink("GET",
```
Comment (garagatyi) [inline core/platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java]: add link inside of java 8 stream processing

**Comment by mmorhun** [comment]
done (as for tests, I built it without skipping of tests and checked work of IDE manually)

File: core/platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java [hunk 3]
```diff
@@ -954,7 +958,24 @@ private void ensureUserIsWorkspaceOwner(UsersWorkspace usersWorkspace) throws Se
...
+
+            if (RuntimeWorkspaceDto.class.isAssignableFrom(workspace.getClass())) {
+                RuntimeWorkspaceDto runtimeWorkspace = (RuntimeWorkspaceDto)workspace;
+                runtimeWorkspace.getDevMachine()
+                                .getRuntime()
+                                .getServers()
+                                .values()
+                                .stream()
+                                .filter(server ->  WS_AGENT_SERVER_REFERENCE.equals(server.getRef()))
+                                .findFirst()
```
Comment (garagatyi) [inline core/platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java]: findAny?

File: core/platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java [hunk 4]
```diff
@@ -954,7 +958,24 @@ private void ensureUserIsWorkspaceOwner(UsersWorkspace usersWorkspace) throws Se
...
+                                .ifPresent(extensionServer -> links.add(createLink("GET",
```
Comment (garagatyi) [inline core/platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java]: wsAgent is modern terminology

File: core/platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/shared/Constants.java [hunk 5]
```diff
@@ -41,6 +41,7 @@
     public static final String LINK_REL_DELETE_ICON = "delete icon link";
 
     public static final String LINK_REL_GET_WORKSPACE_EVENTS_CHANNEL = "get workspace events channel";
+    public static final String LINK_REL_GET_WEB_SOCKET_URL_EXT_SERVER = "get websocket url of extension server";
```
Comment (garagatyi) [inline core/platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/shared/Constants.java]: wsAgent is modern terminology

File: core/platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/shared/Constants.java [hunk 6]
```diff
@@ -41,6 +41,7 @@
     public static final String LINK_REL_DELETE_ICON = "delete icon link";
 
     public static final String LINK_REL_GET_WORKSPACE_EVENTS_CHANNEL = "get workspace events channel";
+    public static final String LINK_REL_GET_WEB_SOCKET_URL_WS_AGENT  = "get websocket url of extension server";
```
Comment (garagatyi) [inline core/platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/shared/Constants.java]: change value to `wsagent.websocket`

**Comment by mmorhun** [comment]
Is it ok?

**Comment by garagatyi** [comment]
LGTM

**Comment by garagatyi** [comment]
@skabashnyuk ?

**Comment by skabashnyuk** [comment]
ok @vparfonov ?

**Comment by vparfonov** [comment]
ok

**Pull request merged**